### ISSUE TITLE
StringIndexOutOfBoundsException when checking grammar and spelling problems

### ISSUE THREAD

**Issue by ahus1** [asciidoctor/asciidoctor-intellij-plugin#752]
Reported by Sentry for plugin version 0.32.44

``` 
java.lang.StringIndexOutOfBoundsException: start 40, end 38, length 38
    at java.lang.AbstractStringBuilder.checkRangeSIOOBE(AbstractStringBuilder.java:1724)
    at java.lang.AbstractStringBuilder.delete(AbstractStringBuilder.java:863)
    at java.lang.StringBuilder.delete(StringBuilder.java:284)
    at com.intellij.grazie.grammar.GrammarChecker.determineTextShifts(GrammarChecker.kt:73)
    at com.intellij.grazie.grammar.GrammarChecker.check(GrammarChecker.kt:93)
    at com.intellij.grazie.ide.inspection.grammar.GrazieInspection$buildVisitor$1.visitElement(GrazieInspection.kt:102)
    at com.intellij.psi.impl.PsiElementBase.accept(PsiElementBase.java:273)
    at org.asciidoc.intellij.psi.AsciiDocStandardBlock.accept(AsciiDocStandardBlock.java:30)
    at com.intellij.codeInspection.InspectionEngine.acceptElements(InspectionEngine.java:65)
    at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.lambda$visitRestElementsAndCleanup$7(LocalInspectionsPass.java:346)
    at com.intellij.util.AstLoadingFilter.lambda$toComputable$2(AstLoadingFilter.java:174)
    at com.intellij.util.AstLoadingFilter.disallowTreeLoading(AstLoadingFilter.java:132)
    at com.intellij.util.AstLoadingFilter.disallowTreeLoading(AstLoadingFilter.java:121)
    at com.intellij.util.AstLoadingFilter.disallowTreeLoading(AstLoadingFilter.java:111)
    at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.lambda$visitRestElementsAndCleanup$10(LocalInspectionsPass.java:346)
    at com.intellij.concurrency.ApplierCompleter.execAndForkSubTasks(ApplierCompleter.java:136)
    at com.intellij.concurrency.ApplierCompleter.execAndForkSubTasks(ApplierCompleter.java:149)
    at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1096)
    at com.intellij.concurrency.ApplierCompleter.lambda$wrapInReadActionAndIndicator$1(ApplierCompleter.java:92)
    at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:688)
    at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:634)
    at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:64)
    at com.intellij.concurrency.ApplierCompleter.wrapInReadActionAndIndicator(ApplierCompleter.java:104)
    at com.intellij.concurrency.ApplierCompleter.lambda$compute$0(ApplierCompleter.java:83)
    at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:167)
    at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:178)
    at com.intellij.concurrency.ApplierCompleter.compute(ApplierCompleter.java:83)
    at java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:746)
    at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
    at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
    at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
    at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
    at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
```

**Comment by ahus1** [comment]
Fixing an off-by-one-bug, also adding more logging for the upcoming 0.32.45. Waiting for more information to be reported.

**Comment by ahus1** [comment]
New version rolled out. Still waiting for a user to also report the attachment that would contain a PSI. The PSI would give some hints for the situation to case this exception.

**Comment by ahus1** [comment]
Turned out to be a regression in #747, fixed in 0.32.46

**Comment by ahus1** [comment]
There is still a problem with an edge case, further investigation necessary.

**Comment by ahus1** [comment]
Another fix, will be part of 0.32.47
